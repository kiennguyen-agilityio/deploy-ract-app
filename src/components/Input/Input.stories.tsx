import type { Meta, StoryObj } from '@storybook/react';
import Input from './Input';

const meta: Meta<typeof Input> = {
  component: Input,
};

type Story = StoryObj<typeof Input>;

export const Primary: Story = {
  args: {
    title: 'Input',
    variant: 'primary',
  },
};

export const secondary: Story = {
  args: {
    title: 'Input',
    variant: 'secondary',
  },
};

export default meta;
