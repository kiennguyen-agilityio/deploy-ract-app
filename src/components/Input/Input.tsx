import React, { ChangeEvent, ReactElement } from 'react';
import './index.css';

export interface InputProps {
  name: string;
  value?: string;
  placeholder?: string;
  title?: string;
  variant: 'primary' | 'secondary';
  type?: 'text' | 'number';
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
}

const Input = ({
  name,
  value,
  placeholder,
  title,
  variant,
  type,
  onChange,
}: InputProps): ReactElement => (
  <>
    {title ? (
      <div className="text-wrapper">
        <label>{title}</label>
        <input
          className={`text-input text-input-${variant}`}
          type={type}
          name={name}
          value={value}
          placeholder={placeholder}
          onChange={onChange}
          required
        />
      </div>
    ) : (
      <input
        className={`text-input text-input-${variant}`}
        type={type}
        name={name}
        value={value}
        placeholder={placeholder}
        onChange={onChange}
        required
      />
    )}
  </>
);

export default React.memo(Input);
