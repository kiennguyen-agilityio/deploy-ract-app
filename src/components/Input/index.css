.text-input-primary {
  border: none;
  border-bottom: 2px solid var(--color-secondary);
  background-color: transparent;
  padding: 5px 0;
  outline: none;
  text-align: center;
  color: var(--color-tertiary);
  height: 40px;
  background-color: transparent;
  border-width: 0 0 2px;
  font-size: var(--font-size-md);
  line-height: var(--line-height-40);
  text-transform: capitalize;
  font-weight: var(--font-weight-light);
}

.text-wrapper {
  position: relative;
  display: flex;
  align-items: center;
}

.text-wrapper label {
  position: absolute;
  top: -12px;
  left: 50%;
  background-color: var(--color-primary);
  padding: 0 5px;
  transform: translateX(-50%);
  color: var(--color-gray);
  font-weight: var(font-weight-regular);
}

.text-input-secondary {
  width: 200px;
  height: 40px;
  border: 2px solid var(--color-secondary);
  border-radius: 4px;
  padding: 5px 10px;
  font-size: 16px;
}

.text-input-secondary:focus {
  outline: none;
}
