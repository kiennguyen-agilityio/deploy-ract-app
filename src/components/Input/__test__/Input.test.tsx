import { RenderResult, act, cleanup, fireEvent, render, screen } from '@testing-library/react';

// Components
import Input, { InputProps } from '../Input';

describe('Testing Input component', () => {
  const handleChange = jest.fn();

  afterEach(() => {
    cleanup();
  });

  const defaultProps = {
    name: 'name',
    placeholder: 'Enter topic...',
  } as InputProps;

  const setup = (overrideProps?: Partial<InputProps>): RenderResult => {
    const props: InputProps = {
      ...defaultProps,
      ...overrideProps,
    };

    return render(<Input {...props} onChange={handleChange} />);
  };

  test('Should render correctly Input component', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('Should call onChange event when change value input', () => {
    render(<Input {...defaultProps} onChange={handleChange} />);

    const input = screen.getByPlaceholderText<HTMLInputElement>('Enter topic...');

    act(() => {
      fireEvent.change(input, {
        target: { value: 'English' },
      });
    });

    expect(handleChange).toBeCalled();
    expect(input).toHaveValue('English');
  });

  it('Should render component with title correctly', () => {
    const { getByText } = render(
      <Input {...defaultProps} title="English" onChange={handleChange} />
    );

    const text = getByText('English');

    expect(text).toBeInTheDocument();
  });
});
