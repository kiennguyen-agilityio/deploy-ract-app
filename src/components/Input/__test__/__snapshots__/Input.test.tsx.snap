// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component Should render correctly Input component 1`] = `
<div>
  <input
    class="text-input text-input-undefined"
    name="name"
    placeholder="Enter topic..."
    required=""
    value=""
  />
</div>
`;
